tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -1170,14 +1171,19 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
     config.set_epilogue(GemmBackendConfig::BIAS);
     std::vector<HloInstruction *> operands(gemm->operands().begin(),
                                            gemm->operands().end());
-    operands.push_back(bias);
-
+    if (gemm->custom_call_target() == kCublasLtMatmulF8CallTarget) {
+      // Matrix and vector bias don't co-exist for FP8 matmul.
+      config.set_beta(0.0);
+      operands[2] = bias;
```

Comment:
The documentation for `CUBLASLT_MATMUL_DESC_BIAS_POINTER` states the bias type must be the "same as matrix D datatype" ([link](https://docs.nvidia.com/cuda/cublas/index.html#cublasltmatmuldescattributes-t)). But here, the bias can be FP16 and D can be FP8, and your added test checks this case. Is the documentation wrong for FP8?

Reply:
The table under cublasLtMatmul() lists all possible combinations of D and bias's type. They're not necessarily the same.